UPGRADING.md

```diff
@@ -0,0 +1,317 @@
+# Clerk Go SDK upgrade guide
+
+The Clerk Go SDK uses [Semantic Versioning](https://semver.org/)
+for tracking all changes to the library.
+
+Semantic versions take the form of `MAJOR.MINOR.PATCH`. Incrementing
+
+- MAJOR version introduces incompatible API changes
+- MINOR version introduces functionality in a backwards compatible manner
+- PATCH version includes backwards compatible bug fixes
+
+This file contains upgrade notes for all `MAJOR` version bumps.
+
+While there are no plans to sunset older versions of the Go SDK,
+the current `MAJOR` version of `clerk-sdk-go` is actively supported.
+
+Bug fixes and security updates may be added to previous versions, but
+there is no guarantee.
+
+## 1.x.x to 2.x.x
+
+The `v2` version of the Clerk Go SDK is a complete rewrite and introduces
+a lot of breaking changes.
+
+### Completely different API
+
+API operations in `v1` of the Clerk Go SDK are organized by service. There are
+different services for each API and all services are properties of a single
+`clerk.Client`.
+
+Let's see a concrete example. Assume that we want to create an organization
+and then list all available organizations.
+
+Here's how we would do it in `v1`.
+
+```go
+import (
+    "github.com/clerkinc/clerk-sdk-go"
+)
+
+client, err := clerk.NewClient("sk_live_XXX")
+if err != nil {
+    // handle error
+}
+
+// Create an organization
+org, err := clerk.Organizations().Create(clerk.CreateOrganizationParams{
```

Comment:
This example should use the client variable right?
```suggestion
org, err := client.Organizations().Create(clerk.CreateOrganizationParams{
```

Reply:
Indeed. Fixed.